### PR TITLE
Extend timespan request to also show historic data

### PR DESCRIPTION
**Pull request by kiliankoe**
This is as of yet untested.

### CODE REVIEW

**Comment by jklmnn** [comment]
The current problem is, that the forecast only shows free (or used?) percentage points while all our other timespan formats (including the dumps) use free lots. We cannot let this be inconsistent (especially since it cannot be determined if it is a timespan from the past or a forecast). So we have two options:
 - change the past timespan to percentages
 - change the forecast to free lots and update the apps

I'd go for the second choice as this is the better data format in my eyes and the app update should be quite easy (in the case of android its simply removing the calculation that reverts percentages into free lots.

**Comment by kiliankoe** [comment]
Ah, I didn't even think about the inconsistency, good point! I'm definitely with you that absolute numbers are superior, but as you say this would make changes in the forecast data and apps necessary, making this a weirdly breaking change. Afaik not many users check the forecast feature, so it's probably fine to cause potential inconsistencies here. I'll check the iOS update tonight.

**Comment by jklmnn** [comment]
To solve the inconsistency and compatibility problem this API will be consistent for forecast and historical data in a yet to specify documented format. It will only be available if `version=1b` is added to the url parameters. Version `2` will not be used to keep it free for the next API.

**Comment by jklmnn** [comment]
The new API will return the following schema:
```JSON
{
  "data": [
    {"timestamp": "2019-09-02T00:00:00",
     "free": "9"}
  ],
  "version": "1b"
}
```

File: park_api/app.py [hunk 1]
```diff
@@ -74,47 +109,54 @@ def get_lots(city):
...
+    except KeyError:
+        data = None
     if data is not None:
-        return jsonify(data)
+        return jsonify({
+            'version': version,
+            'data': data
+        })
     else:
+        print(data)
```
Comment (kiliankoe) [inline park_api/app.py]: This is probably just some leftover printf-debugging :P

File: park_api/timespan.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import csv
+from datetime import datetime
+
+from park_api import db, env
+
+
+def timespan(city, lot_id, total, date_from, date_to, version):
+    now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S")
+    if date_from > now or version == 1.0:
+        print("forecast")
```
Comment (kiliankoe) [inline park_api/timespan.py]: Here as well^^

File: park_api/timespan.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        data = forecast(lot_id, total, date_from, date_to, version)
+    elif date_to < now:
+        print("timespan")
```
Comment (kiliankoe) [inline park_api/timespan.py]: And here.

File: park_api/timespan.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        data = known_timespan_data(city, lot_id, date_from, date_to, version)
+    else:
+        print("both")
```
Comment (kiliankoe) [inline park_api/timespan.py]: 👆

File: park_api/app.py (hunk 1 above)
Comment (kiliankoe) [reply park_api/app.py]:
```suggestion
```

File: park_api/timespan.py (hunk 2 above)
Comment (kiliankoe) [reply park_api/timespan.py]:
```suggestion
```

File: park_api/timespan.py (hunk 3 above)
Comment (kiliankoe) [reply park_api/timespan.py]:
```suggestion
```

File: park_api/timespan.py (hunk 4 above)
Comment (kiliankoe) [reply park_api/timespan.py]:
```suggestion
```

**Pull request merged**